### ISSUE TITLE
[FEATURE] Add defaults to Steps and Tasks so they can be more easily connected

### ISSUE THREAD

**Issue by dvsrepo** [argilla-io/distilabel#802]
This is an issue to discuss the defaults for (some) Steps and Task.

The main idea is to think about the most frequent uses and paths for certain components. I'll start:

1. `CombineColumns`: In my experience, the most likely usage of this is after several generation steps (which by default output `generation` and `model_name`). Would it make sense to set this as default values for `columns` . This could allow to go from:
```python
text_generation >> CombineColumns(columns=["generation", "model_name"])

# to 

text_generation >> CombineColumns()
```

It might not seem as a very impactful change but I think adding these small changes will compound and give a more intuitive and approachable usage/onboarding. The most important is not having shorter code but less things to understand and memorize for repetitive actions.

**Comment by gabrielmbmb** [comment]
Yes, I think it makes sense to have these sensible defaults. Something related I was thinking is to also have auxiliary functions that creates several steps, kind of syntactic sugar:

```python
def generate_with(models: List[str]): ...

with Pipeline(name="my-pipe") as pipeline:
     load_data = ...
     
     load_data >> generate_with(["llama3", "mistral", "gpt4"])
```

but not 100% sure about it because it can hide many details and make the user confuse
